lib/bolt/puppetdb/client.rb

```diff
@@ -95,6 +95,60 @@ def make_query(query, path = nil)
         make_query(query, path)
       end
 
+      # Sends a command to PuppetDB using version 1 of the commands API.
+      # https://puppet.com/docs/puppetdb/latest/api/command/v1/commands.html
+      #
+      # @param command [String] The command to invoke.
+      # @param version [Integer] The version of the command to invoke.
+      # @param payload [Hash] The payload to send with the command.
+      # @return A UUID identifying the submitted command.
+      #
+      def send_command(command, version, payload)
+        command = command.dup.force_encoding('utf-8')
+        body    = JSON.generate(payload)
+
+        # PDB requires the following query parameters to the POST request.
+        # Error early if there's no certname, as PDB does not return a
+        # message indicating it's required.
+        unless payload['certname']
```

Comment:
In testing the endpoint it seems like all keys are required and should be validated either here or in the plan function itself since PDB doesn't raise an error if the request doesn't include them or has the wrong type. While I'm inclined to say this will most likely be used by people who have used the API before, I also think part of Bolt's mission is to make things easy for users new to the Puppet ecosystem for whom the required keys here might not be obvious.

Required:
- environment, string
- producer_timestamp, string
- values, keys are strings
- optional producer, if not provided add it automatically as 'nil' (since it's required by pdb)